pkg/api/controller.go

```diff
@@ -2845,6 +2846,53 @@ func (c *Controller) logCommitsHelper(w http.ResponseWriter, r *http.Request, re
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) HeadObject(w http.ResponseWriter, r *http.Request, repository string, ref string, params HeadObjectParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ReadObjectAction,
+			Resource: permissions.ObjectArn(repository, params.Path),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "head_object", r, repository, ref, "")
+
+	// read the FS entry
+	entry, err := c.Catalog.GetEntry(ctx, repository, ref, params.Path, catalog.GetEntryParams{ReturnExpired: true})
+	rr := httptest.NewRecorder()
+	if c.handleAPIError(ctx, w, err) {
+		w.WriteHeader(rr.Code)
+		return
+	}
+	c.Logger.Tracef("get repo %s ref %s path %s: %+v", repository, ref, params.Path, entry)
```

Comment:
The tracing middleware should give you the same information. In case it doesn't - please use the logger's WithFields to capture the structured data.

Reply:
Removed.